modules/local/validation/metadata/validate_metadata.py

```diff
@@ -0,0 +1,268 @@
+    read_group_files = set()
+    
+    for rg in read_groups:
+        rg_id = rg.get('submitter_read_group_id', 'unknown')
+        
+        # Check file_r1
+        file_r1 = rg.get('file_r1', '').strip()
+        if file_r1:
+            read_group_files.add(file_r1)
+            if file_r1 not in payload_filenames:
+                validation_errors.append(f'Read group {rg_id}: file_r1 "{file_r1}" not found in payload files')
+        
+        # Check file_r2
+        file_r2 = rg.get('file_r2', '').strip()
+        if file_r2:
+            read_group_files.add(file_r2)
+            if file_r2 not in payload_filenames:
+                validation_errors.append(f'Read group {rg_id}: file_r2 "{file_r2}" not found in payload files')
+        
+        # Validate library layout consistency
+        library_layout = rg.get('library_layout', '').upper()
+        if library_layout == 'PAIRED':
+            if not file_r1:
+                validation_errors.append(f'Read group {rg_id}: library_layout is PAIRED but file_r1 is missing')
+            if not file_r2:
+                validation_errors.append(f'Read group {rg_id}: library_layout is PAIRED but file_r2 is missing')
+        elif library_layout == 'SINGLE':
+            if not file_r1:
+                validation_errors.append(f'Read group {rg_id}: library_layout is SINGLE but file_r1 is missing')
+            if file_r2:
+                validation_errors.append(f'Read group {rg_id}: library_layout is SINGLE but file_r2 is provided')
+        elif library_layout and library_layout not in ['PAIRED', 'SINGLE']:
+            validation_errors.append(f'Read group {rg_id}: invalid library_layout "{library_layout}", must be PAIRED or SINGLE')
+    
+    print(f'Found {len(read_group_files)} unique files referenced in read groups: {sorted(read_group_files)}')
+    
+    # Check for payload files not referenced in read groups
+    unreferenced_files = payload_filenames - read_group_files
+    if unreferenced_files:
+        for uf in sorted(unreferenced_files):
+            validation_errors.append(f'Payload file "{uf}" not referenced in any read group')
+    
+    return validation_errors
+
+
+def validate_read_group_fields(read_groups):
+    """Validate required fields in read groups."""
+    validation_errors = []
+    
+    # Check for additional required read group fields
+    required_rg_fields = ['submitter_read_group_id', 'submitter_experiment_id', 'library_name', 'library_layout']
+    
+    for i, rg in enumerate(read_groups):
+        rg_id = rg.get('submitter_read_group_id', f'read_group_{i+1}')
+        
+        for field in required_rg_fields:
+            if not rg.get(field, '').strip():
+                validation_errors.append(f'Read group {rg_id}: missing required field "{field}"')
+        
+        # Additional validation for specific fields
+        platform_unit = rg.get('platform_unit', '').strip()
+        if platform_unit and len(platform_unit) > 100:
```

Comment:
Do we want to add a character limit in lectern for `platform_unit`
https://github.com/Pan-Canadian-Genome-Library/data-dictionary/blob/c6266223e2b594fa514a1d56675e56cbd865b1c2/lectern/pcgl/pcgl_lectern.json#L2800-L2813

Reply:
Not addressed b/c no longer needed